### PR TITLE
Promote healthcare resources to GA

### PR DESCRIPTION
**Pull request by umairidris**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_healthcare_dataset` is now GA
```
```release-note:new-resource
`google_healthcare_dicom_store` is now GA
```
```release-note:new-resource
`google_healthcare_fhir_store` is now GA
```
```release-note:new-resource
`google_healthcare_hl7_v2_store` is now GA
```

### CODE REVIEW

**Comment by danawillow** [comment]
(PR comes from a googler so I rolled a metaphorical die)

File: third_party/terraform/tests/resource_healthcare_fhir_store_test.go.erb [hunk 1]
```diff
@@ -162,33 +162,6 @@ resource "google_pubsub_topic" "topic" {
 `, fhirStoreName, datasetName, pubsubTopic)
 }
 
-func testAccCheckHealthcareFhirStoreDestroyProducer(t *testing.T) func(s *terraform.State) error {
```
Comment (chrisst) [inline third_party/terraform/tests/resource_healthcare_fhir_store_test.go.erb:165]: I don't think you meant to remove these.

File: third_party/terraform/tests/resource_healthcare_dataset_test.go.erb [hunk 2]
```diff
@@ -106,33 +106,6 @@ func TestAccHealthcareDataset_basic(t *testing.T) {
 	})
 }
```
Comment (chrisst) [inline third_party/terraform/tests/resource_healthcare_dataset_test.go.erb:108]: These test files can be swapped from being erb's to straight `.go` files. They are only templated files so they can ben excluded from the GA provider.

Comment (umairidris) [reply third_party/terraform/tests/resource_healthcare_dataset_test.go.erb:108]: Done.

**Comment by umairidris** [comment]
Addressed comments, PTAL!

**Comment by umairidris** [comment]
@chrisst it would be great if we could get this in time for the next release. Are there any pending items? I can't quite see what the trigger failure is but I imagine it's because of the release note missing?

**Comment by chrisst** [comment]
builds are passing 👍 I ran a subset of tests locally and they are passing too.

**Pull request merged**